src/compare-pdf-to-snapshot.test.ts

```diff
@@ -193,4 +193,68 @@ describe('comparePdfToSnapshot()', () => {
         .then(() => unlinkSync(snapshotPath))
     })
   })
+
+  describe('when previous snapshot file does not exist', () => {
+    describe('allowSnapshotCreation is true (default)', () => {
+      it('should create new snapshot if missing and failOnMissingSnapshot is false (default)', () => {
+        const snapshotName = 'allow-create-snapshot'
+        const snapshotPath = join(__dirname, snapshotsDirName, snapshotName + '.png')
+        if (existsSync(snapshotPath)) {
+          unlinkSync(snapshotPath)
+        }
+        return comparePdfToSnapshot(singlePageSmallPdfPath, __dirname, snapshotName).then((x) => {
+          assert.strictEqual(x, true)
+          assert.strictEqual(existsSync(snapshotPath), true, 'Snapshot should be created')
+          unlinkSync(snapshotPath)
+        })
+      })
+
+      it('should create new snapshot if missing and failOnMissingSnapshot is true', () => {
+        const snapshotName = 'allow-create-snapshot'
+        const snapshotPath = join(__dirname, snapshotsDirName, snapshotName + '.png')
+        if (existsSync(snapshotPath)) {
+          unlinkSync(snapshotPath)
+        }
+        return comparePdfToSnapshot(singlePageSmallPdfPath, __dirname, snapshotName, {
+          failOnMissingSnapshot: true,
+        }).then((x) => {
+          assert.strictEqual(x, true)
+          assert.strictEqual(existsSync(snapshotPath), true, 'Snapshot should be created')
+          unlinkSync(snapshotPath)
+        })
+      })
+    })
+
+    describe('allowSnapshotCreation is false', () => {
+      it('should fail if snapshot is missing and failOnMissingSnapshot is true', () => {
+        const snapshotName = 'fail-on-missing-snapshot'
+        const snapshotPath = join(__dirname, snapshotsDirName, snapshotName + '.png')
+        if (existsSync(snapshotPath)) {
+          unlinkSync(snapshotPath)
+        }
+        return comparePdfToSnapshot(singlePageSmallPdfPath, __dirname, snapshotName, {
+          allowSnapshotCreation: false,
+          failOnMissingSnapshot: true,
+        }).then((x) => {
+          assert.strictEqual(x, false)
+          assert.strictEqual(existsSync(snapshotPath), false, 'Snapshot should not be created')
+        })
+      })
+
+      it('should pass if snapshot is missing and failOnMissingSnapshot is false (default)', () => {
+        const snapshotName = 'pass-no-create-no-fail'
+        const snapshotPath = join(__dirname, snapshotsDirName, snapshotName + '.png')
+        if (existsSync(snapshotPath)) {
+          unlinkSync(snapshotPath)
+        }
+        return comparePdfToSnapshot(singlePageSmallPdfPath, __dirname, snapshotName, {
+          allowSnapshotCreation: false,
```

Comment:
@moshensky these specs look like they cover the new options and functionality but I was wondering if you think we should also check the interaction with the existing [approve](https://github.com/moshensky/pdf-visual-diff/blob/master/src/cli/approve.ts) and discard CLI commands?

From what I can see the approve and discard CLI command module is entirely isolated from the compare pdf command (which would be run as part of people's test suite) so it shouldn't be an issue but I'm a bit unclear on what happens if `allowSnapshotCreation` is `false` and there's an existing snapshot, I assume the `.diff.png` would be created but not the `.new.png`?

If that's so maybe dropping a hint in the readme that to use the `test:pdf-approve` command allow snapshot creation needs to be true while running specs -- or the opposite i.e. make it clear that `allowSnapshotCreation` has no effect on the .diff.pdf and .new.pdf output.